bundles/org.eclipse.swt/Eclipse SWT/win32/org/eclipse/swt/widgets/Shell.java

```diff
@@ -298,8 +300,28 @@ public Shell (Display display, int style) {
 	if (handle != 0 && !embedded) {
 		state |= FOREIGN_HANDLE;
 	}
+
+	int [] dpiX = new int [1];
+	int [] dpiY = new int [1];
+	int shellZoomFactor;
+	int shellNativeZoomFactor;
+	if (parent != null) {
+		shellZoomFactor = parent.getZoomFactor();
+		shellNativeZoomFactor = parent.getNativeZoomFactor();
+	} else {
+		Monitor monitor = getMonitor();
+		OS.GetDpiForMonitor (monitor.handle, 0, dpiX, dpiY);
```

Comment:
This method was introduced with Windows 8.1 and will thus not return usable results for older Windows versions: https://learn.microsoft.com/en-US/windows/win32/api/shellscalingapi/nf-shellscalingapi-getdpiformonitor

You can have a look at `Display.getMonitor(long)`, which also uses that method.

Reply:
see below